python-runnables/resize-node-pool/runnable.py

```diff
@@ -42,20 +42,45 @@ def run(self, progress_callback):
         node_pool_id = node_pool.name
         logging.info("Node pool selected is %s " % node_pool_id)
 
-        desired_count = self.config['numNodes']
-        logging.info("Resize to %s" % desired_count)
-        if desired_count == 0:
-            if len(node_pools) == 1:
-                raise Exception("Can't delete node pool, a cluster needs at least one running node pool")
-            def do_delete():
-                cluster_update_op = clusters.agent_pools.begin_delete(resource_group, cluster_name, node_pool_id)
-                return cluster_update_op.result()
-            delete_result = run_and_process_cloud_error(do_delete)
-            logging.info("Cluster updated")
-            return '<pre class="debug">Node pool %s deleted</pre>' % node_pool_id
+        autoscaling_enabled = self.config['autoScaling']
+        node_pool.enable_auto_scaling = autoscaling_enabled
+        if autoscaling_enabled:
+            min_nodes = self.config['minNumNodes']
+            max_nodes = self.config['maxNumNodes']
+            initial_node_count = self.config['numNodes']
+            if min_nodes is None:
+                raise Exception("Cannot make autoscalable cluster with no minimum number of nodes.")
+            elif max_nodes is None:
+                raise Exception("Cannot make autoscalable cluster with no maximum number of nodes.")
+            elif initial_node_count is None or initial_node_count < min_nodes or initial_node_count > max_nodes:
+                raise Exception("The initial number of nodes must be set and between the min and max number of nodes bounds ([%s %s])."
+                                % (min_nodes, max_nodes))
+            elif min_nodes > max_nodes:
+                raise Exception("Cannot make autoscalable cluster with a max number of nodes (%s) less than its min number of nodes (%s)."
+                                % (max_nodes, min_nodes))
+            else:
+                logging.info("Resizing cluster to autoscale with %s min nodes and %s max nodes and an initial node count of %s."
+                             % (min_nodes, max_nodes, initial_node_count))
+                node_pool.min_count = min_nodes
+                node_pool.count = initial_node_count
+                node_pool.max_count = max_nodes
+                # This is used in the autoscaling node pool allocation, is it necessary?
```

Comment:
We need to figure that out. We want VirtualMachineScaleSets and not AvailabilitySet.
I think the default is VirtualMachineScaleSets right? So I think we could remove that (this way if Azure introduce a better default we will have it without having to think about it)

Reply:
Not sure what te default is, but I added this for consistency with the node pool creation where we set it to VirtualMachinesScalesSet. I haven't found evidence in the SDK documentation about what is the default value.
However, I am going to remove this because if the node pool was created with a certain type, we probably don't want to change it without the user knowing about it.